### PR TITLE
Don't allow XSS on title in Chicago Formatter

### PR DESCRIPTION
**Pull request by no-reply**
There's probably a larger fix related to forms here, but the chicago formatter
needn't be so permissive.

Related to #3187.

Acceptance: Check XSS referenced in #3187, **Vulnerability 1**

@samvera/hyrax-code-reviewers

### CODE REVIEW

**Comment by no-reply** [comment]
> If Chicago Formatter is returning a string of HTML, shouldn’t something downstream notice it’s not Html_safe? and do the escape?

**Comment by no-reply** [comment]
@jcoyne:

> If Chicago Formatter is returning a string of HTML, shouldn’t something downstream notice it’s not Html_safe? and do the escape?

Yes I think so. This is clearly not the case, however, and the issue seems not to be isolated to the citation formatters. Pending a more complete fix, this patches a known XSS vulnerability.

**Comment by no-reply** [comment]
@jcoyne after a closer look, this formatter marks its output `#html_safe`, which was the real source of the issue.

I've added whitewashing for the remainder of the fragments compiled into the formatted output.

File: app/helpers/hyrax/citations_behaviors/formatters/chicago_formatter.rb [hunk 1]
```diff
@@ -13,11 +13,11 @@ def format(work)
           text = "<span class=\"citation-author\">#{text}</span>" if text.present?
           # Get Pub Date
           pub_date = setup_pub_date(work)
-          text << " #{pub_date}." unless pub_date.nil?
+          text << " #{whitewash(pub_date)}." unless pub_date.nil?
```
Comment (jcoyne) [inline app/helpers/hyrax/citations_behaviors/formatters/chicago_formatter.rb]: Can we use `sanitize` in the helpers?  That seems to be the built in fix.

**Pull request merged**